task3_snkrs/spiders/orsay_spider.py

```diff
@@ -0,0 +1,109 @@
+import math
+import scrapy
+from task3_snkrs.items import Item
+
+
+class OrsaySpider(scrapy.Spider):
+    name = "orsay"
+    start_urls = [
+        'https://www.orsay.com/de-de/produkte/'
+    ]
+
+    def parse(self, response):
+        urls_s = response.css('a.refinement-category-link')
+
+        return [response.follow(url_s, callback=self.parse_categories) for url_s in urls_s]
+
+    def parse_categories(self, response):
+        urls_s = response.css('a.refinement-category-link')
+
+        return [response.follow(url_s, callback=self.parse_products_url) for url_s in urls_s]
+
+    def parse_products_url(self, response):
+        urls_s = response.css('a.thumb-link')
+        yield from [response.follow(url_s, callback=self.parse_product) for url_s in urls_s]
+
+        products_found = int(self.get_total_products(response))
+        if products_found > 72:
+            pages = math.ceil(products_found / 72) + 1
+            load_more = '?sz={}'
+            return [response.follow(load_more.format(page * 72), callback=self.parse_products_url)
+                    for page in range(2, pages)]
+
+    def parse_product(self, response):
+        item = Item()
+        item['item_id'] = self.get_id(response)
+        item['name'] = self.get_name(response)
+        item['description'] = self.get_description(response)
+        item['availability'] = self.get_availability(response)
+        item['currency'] = self.get_currency(response)
+        item['skus'] = [self.get_skus(response)]
+        color_urls = self.get_color_wise_urls(response)
+
+        return item if not color_urls else self.color_request(color_urls, item)
+
+    def color_request(self, color_urls, item):
+        return scrapy.Request(color_urls.pop(), callback=self.parse_sizes, meta={'item': item, 'urls': color_urls})
+
+    def parse_sizes(self, response):
```

Comment:
Let's define `call back methods` at the top of all methods of the class (scrapy docs).